packages/web/src/components/upload-confirmation-modal/UploadConfirmationModal.tsx

```diff
@@ -0,0 +1,90 @@
+import { useCallback } from 'react'
+
+import { uploadConfirmationModalUISelectors } from '@audius/common'
+import {
+  Button,
+  ButtonType,
+  IconUpload,
+  Modal,
+  ModalContent,
+  ModalContentText,
+  ModalFooter,
+  ModalHeader,
+  ModalTitle
+} from '@audius/stems'
+
+import { useModalState } from 'common/hooks/useModalState'
+import { useSelector } from 'common/hooks/useSelector'
+import { Text } from 'components/typography'
+
+import styles from './UploadConfirmationModal.module.css'
+
+const { getConfirmCallback, getHasPublicTracks } =
+  uploadConfirmationModalUISelectors
+
+const messages = {
+  title: 'Confirm Upload',
+  publicDescription:
+    'Ready to begin uploading? Your followers will be notified once your upload is complete.',
+  hiddenDescription: 'Ready to begin uploading?',
+  cancel: 'Go Back',
+  upload: 'Upload'
+}
+
+export const UploadConfirmationModal = () => {
+  const confirmCallback = useSelector(getConfirmCallback)
+  const hasPublicTracks = useSelector(getHasPublicTracks)
+  const [isOpen, setIsOpen] = useModalState('UploadConfirmation')
```

Comment:
should we make the modal names constants somewhere?

Reply:
They are defined in the Modals type and then auto completed everywhere else

Reply:
Ohh neat!